### ISSUE TITLE
IOR 3.2.0 with beegfsNumTargets and single file (no -F option)

### ISSUE THREAD

**Issue by k0** [hpc/ior#141]
hi.

while running IOR 3.2.0 on beegfs-7.1.2 with the option -O beegfsNumTargets=n (any n) and using a single output file (no -F option), i get an error:

`ior ERROR: Could not create file, errno 17, File exists (aiori-POSIX.c:251)`

in the file aiori-POSIX.c:251 there is a call to the beegfs API:
`beegfs_createFile(parentDirFd, filename, mode, numTargets, chunkSize);`

but it seems to me that this function gives an error, in the general case, if the file preexists, as can see by compiling and running the example at:
[api example: create file with pattern](https://www.beegfs.io/wiki/StripingAPI#example_create_pattern)

i suppose that the file in the IOR test preexists to other mpi process as soon as it is created by the first process, then the error... ?

with the -F option in the IOR command everithing is OK

is there something i am doing wrong? 

--------------------------------------------------------------------------

log:
```

$ mpirun -np 3 --map-by node -hostfile /beegfs/ior-3.2.0/k0build/bin/hostfile /beegfs/ior-3.2.0/k0build/bin/ior  -o /beegfs/ior-3.2.0/k0build/tests/test/ior-test.file -O beegfsNumTargets=1

[[8646,1],1]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)

Another transport will be used instead, although this may result in
lower performance.

IOR-3.2.0: MPI Coordinated Test of Parallel I/O
Began               : Wed Apr  3 14:42:10 2019
Command line        : /beegfs/ior-3.2.0/k0build/bin/ior -o /beegfs/ior-3.2.0/k0build/tests/test/ior-test.file -O beegfsNumTargets=1
Machine             : Linux 
TestID              : 0
StartTime           : Wed Apr  3 14:42:10 2019
Path                : /beegfs/ior-3.2.0/k0build/tests/test
FS                  : 787.2 GiB   Used FS: 33.0%   Inodes: 0.0 Mi   Used Inodes: -nan%

Options: 
api                 : POSIX
apiVersion          : 
test filename       : /beegfs/ior-3.2.0/k0build/tests/test/ior-test.file
access              : single-shared-file
type                : independent
segments            : 1
ordering in a file  : sequential
ordering inter file : no tasks offsets
tasks               : 3
clients per node    : 1
repetitions         : 1
xfersize            : 262144 bytes
blocksize           : 1 MiB
aggregate filesize  : 3 MiB

Results: 

access    bw(MiB/s)  block(KiB) xfer(KiB)  open(s)    wr/rd(s)   close(s)   total(s)   iter

MPI_ABORT was invoked on rank 2 in communicator MPI_COMM_WORLD 
with errorcode -1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

ior ERROR: Could not create file, errno 17, File exists (aiori-POSIX.c:251)
ior ERROR: Could not create file, errno 17, File exists (aiori-POSIX.c:251)
```

best.

k0

**Comment by glennklockwood** [comment]
This is going to be tough to debug since I don't think any of the core IOR team has access to a BeeGFS system.

@osteffen, I think you contributed this code.  Any chance you could take a look?

**Comment by osteffen** [comment]
Hi everyone.
I can have a look.

(Sorry for the late response, I was on vacation.)

**Comment by osteffen** [comment]
Fixed by #145

**Comment by glennklockwood** [comment]
Thanks!  Looks good and merged it into master.
